### PR TITLE
DDT-330: show timestamp types based on publisher role

### PR DESCRIPTION
**Pull request by mp-assco**

### CODE REVIEW

File: ui/src/app/model/enums/publisherRole.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
...
-    PLT = "PLT",
-    TR = "TR"
-  }
\ No newline at end of file
+  CA = "CA",    // Carrier
+  AG = "AG",    // ???
+  VSL = "VSL",  // Vessel?
+  ATH = "ATH",  // Port Authority
+  PLT = "PLT",  // Pilot
+  TR = "TR"     // Terminal
```
Comment (nt-as) [inline ui/src/app/model/enums/publisherRole.ts:7]: According to the Bussiness Requirements, there are the following 6 roles:


Carrier:
o	OPS for Carrier Operations  (I assume this is what we know as `CA`)
o	AGE for Carrier local Agent (I assume this is what we know as `AG`)
o	VSL for Vessel
Port:
o	ATH for Port Authorities
o	PLT for Port Pilot
Terminal: 
o	TRM for Terminal (I assume we use `TR` here)

File: ui/src/app/model/portCall/portcall-timestamp-type.enum.ts [hunk 2]
```diff
@@ -18,3 +20,43 @@ export enum PortcallTimestampType {
   PTD_Berth = 'PTD-Berth',
   ATD_Berth = 'ATD-Berth'
 }
+
+export class TimestampTypeUtil {
+
+  static getPortcallTimestampTypes(publisherRole: PublisherRole): string[] {
+    switch (publisherRole) {
+      case PublisherRole.CA:
```
Comment (nt-as) [inline ui/src/app/model/portCall/portcall-timestamp-type.enum.ts:28]: I assume `AG` and `VSL` should go here too.  If in doubt, feel free to ask Guido.

File: ui/src/app/model/portCall/portcall-timestamp-type.enum.ts [hunk 3]
```diff
@@ -18,3 +20,43 @@ export enum PortcallTimestampType {
...
+          PortcallTimestampType.RTD_Berth,
+        ]
+      case PublisherRole.PLT:
+        return [
+          PortcallTimestampType.ATS_Pilot,
+        ]
+      default:
+        return [];
+    }
+  }
```
Comment (nt-as) [inline ui/src/app/model/portCall/portcall-timestamp-type.enum.ts:61]: @AndreasHoff:  Could you please go through this and match it with the most recent table we got from Guido? (To confirm that those rules are the `Sender` of those timestamps).

(It looks correct at first glance)

File: ui/src/app/model/portCall/portcall-timestamp-type.enum.ts (hunk 2 above)
Comment (mp-assco) [reply ui/src/app/model/portCall/portcall-timestamp-type.enum.ts:28]: I talked with Charles Moret regarding roles and according to the table he sent me it doesn't seem like the remaining roles can create timestamps in the JIT (or is something we don't have to implement at the moment).

Comment (nt-as) [reply ui/src/app/model/portCall/portcall-timestamp-type.enum.ts:28]: Charles is as good an authority as Guido! Thanks for checking.

File: ui/src/app/model/enums/publisherRole.ts (hunk 1 above)
Comment (nt-as) [reply ui/src/app/model/enums/publisherRole.ts:7]: (so we can replace the `?`s with better comments)

File: ui/src/app/model/portCall/portcall-timestamp-type.enum.ts (hunk 3 above)
Comment (AndreasHoff) [reply ui/src/app/model/portCall/portcall-timestamp-type.enum.ts:61]: Everything seems to be in order @nt-as ! :)

File: ui/src/app/model/enums/publisherRole.ts (hunk 1 above)
Comment (mp-assco) [reply ui/src/app/model/enums/publisherRole.ts]: @nt-as done!

**Review by nt-as — APPROVED** [review]

**Pull request merged**